opendm/config.py

```diff
@@ -536,7 +536,7 @@ def config(argv=None, parser=None):
                         metavar='<float>',
                         action=StoreValue,
                         type=float,
-                        default=5,
+                        default=10,
```

Comment:
I'm not sure I understand; why is this being changed?

Reply:
Perhaps I am wrong on the current logic, but my assumption is that if someone runs something with e.g. dsm+dtm, they'll get an orthophoto at the default 5cm and their DEM products at 10cm because of the existing gsd_scaling variable. If folks are used to these defaults and depend on them to avoid crashing with their available memory, we might not want to change that behavior.

Reply:
So there's multiple things to consider (which I think might have not been considered here and #1608 ):
 * `gsd_scaling` applies only in regard to the *maximum* allowed resolution per GSD cap
 * `--dem-resolution` (capped, sometimes by an estimate that uses `gsd_scaling`) always affects the output raster resolution (width of the raster cells). If you want ortho+dtm+dsm to have the same raster resolution, they need to be both `5`. Changing it to `10` will decrease your DEMs resolution by half.
 * `--dem-resolution` *also* affects the search radius for binning points. See `odm_dem.py` and `radius_steps` for how that works.

Reply:
I'm not sure how this changes my understanding, but perhaps there are still elements I am missing. Let's dive in and find out what I'm missing:

> gsd_scaling applies only in regard to the maximum allowed resolution per GSD cap

Understood. It is the cap we are asserting needs changed. If I fly a photo set at 1cm GSD, set my orthophoto resolution to 1cm or finer, and ODM estimates ~1cm GSD, I would like to have the capacity without using `ignore-gsd` to set my DEM resolution to the maximum available resolution.

> --dem-resolution (capped, sometimes by an estimate that uses gsd_scaling) always sets the output raster resolution (width of the raster cells). If you want ortho+dtm+dsm to have the same raster resolution, they need to be both 5. Changing it to 10 will decrease your DEMs resolution by half.

This is the current default behavior as I understand it with datasets of 5cm: orthophoto ends up 5cm and DEM ends up 10cm because it currently hits the gsd_scaling cap. But, thinking about it further, changing the defaults only makes sense for that narrow set of cases, so I will revert this change in the pull request.

> --dem-resolution also affects the search radius for binning points. See odm_dem.py and radius_steps for how that works.

I am familiar, and have tuned those parameters for specific datasets when necessary, usually after post-processing the point cloud by using pc2dem.py for fussier elevation model datasets.

> Maybe the real intent is to tweak/decrease the radius_steps, which can sometimes bring on more details if your point cloud is dense enough, but not always, it's not a perfect science and benchmarks should be run on a variety of datasets (with different point densities / capture altitude) to establish if improvements are a net-positive across datasets.

The intent is to match the current state of the art with respect to available resolution, which given our current quality of point clouds is IMO attainable. Tuning of radius steps is already user configurable, and the defaults seem generally fine. Now, if we need to change those in response to resolution cap changes then that's a different set of constraints, and we need to finally have benchmark datasets sufficient in variety. Which... that's on my list.

Reply:
> changing the defaults only makes sense for that narrow set of cases 

:+1: 

> given our current quality of point clouds 

Well, I think it depends on the settings. For example, this is brighton beach (master) run with:

```` 
./run.sh /datasets/brighton --rerun odm_dem --dem-resolution 0.00001 --dsm
```

![image](https://user-images.githubusercontent.com/1951843/219974478-f523716a-c302-4d72-bf75-b7b61a9f68ea.png)


And with the proposed patch:

![image](https://user-images.githubusercontent.com/1951843/219974495-339e1718-10a7-4864-877c-767e21040479.png)

Now, if I also double the `radius_steps` I get a better result:

![image](https://user-images.githubusercontent.com/1951843/219974605-81daaf93-45c1-4879-a423-e709eeba41c1.png)

But is it really?

master:

![image](https://user-images.githubusercontent.com/1951843/219974635-cdd3a754-2cc2-4bae-b6d1-51753fd77d34.png)

patched (+ radius_steps increase):

![image](https://user-images.githubusercontent.com/1951843/219974621-b058a20e-6c81-4d9b-9a8c-b3c4c65dbf42.png)

The point being: I think we should totally work toward setting `gsd_scaling` to `1`, but I don't think we can/should just change `2` to `1` and be done with it.

Reply:
A possible approach might involve the computation of actual average point spacing in the point cloud to set a better value to radius_steps, rather than using output resolution as a starting point.

Reply:
Hmm. Agreed this is a good counter point, and building logic for discerning point cloud density makes sense, but I think diving into radius steps is the wrong end to tackle first. I think actually tying to `pc-quality` first before modifying radius steps is a good first step. `gsd_scaling` of 1 on both sides here, both `pc-quality` of `ultra`. The one on the left has some extra radius steps that make nominal improvements:

![image](https://user-images.githubusercontent.com/1174901/219979668-86280da5-1782-473b-913b-41d614be794a.png)

Propose, in advance of a method for discerning point density, we test with different values for gsd_scaling that are dependent on pc-quality that roughly replicate what exists, but raises the ceiling at higher pc-quality values.

Reply:
Modified the pull request to only allow gsd_scaling of 1.0 with Ultra.